UPGRADING.md

```diff
@@ -0,0 +1,276 @@
+
+The `client.read_rows()` method does not accept `None` anymore as a valid value
+for the optional `metadata` argument. If not given, an empty tuple is used, but
+if you want to explicitly pass an "empty" value, you should use an empty tuple, too.
+
+**Before:**
+```py
+client.read_rows("stream_name", metadata=None)
+```
+
+**After:**
+```py
+client.read_rows("stream_name", metadata=())
+```
+
+
+## Method Calls
+
+> **WARNING**: Breaking change
+
+Most of the client methods that send requests to the backend expect request objects.
+We provide a script that will convert most common use cases.
+
+> One exception to this is the `BigQueryReadClient.read_rows()` which is a hand-written
+wrapper around the auto-generated `read_rows()` method.
+
+* Install the library
+
+```py
+python3 -m pip install google-cloud-bigquery-storage
+```
+
+* The script `fixup_storage_v1_keywords.py` is shipped with the library. It expects
+an input directory (with the code to convert) and an empty destination directory.
+
+```sh
+$ scripts/fixup_storage_v1_keywords.py --input-directory .samples/ --output-directory samples/
+```
+
+**Before:**
+```py
+from google.cloud import bigquery_storage_v1
+
+client = bigquery_storage_v1.BigQueryReadClient()
+
+requested_session = bigquery_storage_v1.types.ReadSession()
+requested_session.table = "projects/PROJECT_ID/datasets/DATASET_ID/tables/TABLE_ID"
+requested_session.data_format = bigquery_storage_v1.enums.DataFormat.ARROW
+
+session = client.create_read_session(
+    "projects/parent_project",
+    requested_session,
+    max_stream_count=1,
+)
+```
+
+**After:**
+```py
+from google.cloud.bigquery import storage
+
+client = storage.BigQueryReadClient()
+
+requested_session = storage.types.ReadSession(
+    table="projects/PROJECT_ID/datasets/DATASET_ID/tables/TABLE_ID",
+    data_format=storage.types.DataFormat.ARROW,
+)
+session = client.create_read_session(
+    request={
+        "parent": "projects/parent_project",
+        "read_session": requested_session,
+        "max_stream_count" 1,
+    },
+)
+```
+
+### More Details
+
+In `google-cloud-bigquery-storage<2.0.0`, parameters required by the API were positional
+parameters and optional parameters were keyword parameters.
+
+**Before:**
+```py
+def create_read_session(
+    self,
+    parent,
+    read_session,
+    max_stream_count=None,
+    retry=google.api_core.gapic_v1.method.DEFAULT,
+    timeout=google.api_core.gapic_v1.method.DEFAULT,
+    metadata=None,
+):
+```
+
+In the `2.0.0` release, methods that interact with the backend have a single
+positional parameter `request`. Method docstrings indicate whether a parameter is
+required or optional.
+
+Some methods have additional keyword only parameters. The available parameters depend
+on the [`google.api.method_signature` annotation](https://github.com/googleapis/python-bigquery-storage/blob/master/google/cloud/bigquery_storage_v1/proto/storage.proto#L73)
```

Comment:
Best to pin to a specific commit in case the file changes in the future and this ends up pointing at the wrong line. (Tip, press the "y" key in GitHub and it changes the URL to the one matching the current commit)

```suggestion
on the [`google.api.method_signature` annotation](https://github.com/googleapis/python-bigquery-storage/blob/9e1bf910e6f5010f479cf4592e25c3b3eebb456d/google/cloud/bigquery_storage_v1/proto/storage.proto#L73)
```

Reply:
Handy!